crates/networking/p2p/rlpx/eth/blocks.rs

```diff
@@ -186,14 +186,33 @@ impl RLPxMessage for BlockHeaders {
 pub(crate) struct GetBlockBodies {
     // id is a u64 chosen by the requesting peer, the responding peer must mirror the value for the response
     // https://github.com/ethereum/devp2p/blob/master/caps/eth.md#protocol-messages
-    id: u64,
-    block_hashes: Vec<BlockHash>,
+    pub id: u64,
+    pub block_hashes: Vec<BlockHash>,
 }
 
+// Limit taken from here:
+// https://github.com/ethereum/go-ethereum/blob/a1093d98eb3260f2abf340903c2d968b2b891c11/eth/protocols/eth/handler.go#L45
+pub const BLOCK_BODY_LIMIT: usize = 1024;
+
 impl GetBlockBodies {
     pub fn new(id: u64, block_hashes: Vec<BlockHash>) -> Self {
         Self { block_hashes, id }
     }
+    pub fn fetch_blocks(&self, storage: &Store) -> Vec<BlockBody> {
+        let mut block_bodies = vec![];
+        for block_hash in &self.block_hashes {
+            let Some(block_body) = storage.get_block_body_by_hash(*block_hash).ok().flatten()
+            else {
+                break;
+            };
```

Comment:
We are hiding internal errors here, if we have problems accessing the state we should return an error instead of acting as if we don't have the block stored

Reply:
You're right, but my doubt here is: what should happen if p2p petition fails? Do we return an empty message? Do we try to fetch more headers in this case? I think a good starting point is to log the error.

Reply:
I'd personally do the following:
1. Log the error.
2. Return empty response. I wouldn't return a partially true response.
3. Add a `TODO(#issue number)` with issue to do research on error handling for eth wire protocol.